nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/users/state/user-listing/user-listing.effects.ts

```diff
@@ -0,0 +1,731 @@
+                                            request: {
+                                                revision: response.revision,
+                                                id: request.user.id,
+                                                uri: request.user.uri,
+                                                userPayload: {
+                                                    ...request.user.component,
+                                                    ...response.user.payload
+                                                }
+                                            }
+                                        })
+                                    );
+                                } else {
+                                    this.store.dispatch(
+                                        UserListingActions.updateUser({
+                                            request: {
+                                                revision: response.revision,
+                                                id: request.user.id,
+                                                uri: request.user.uri,
+                                                userPayload: {
+                                                    ...request.user.component,
+                                                    ...response.user.payload
+                                                },
+                                                userGroupUpdate: {
+                                                    requestId: this.requestId++,
+                                                    userGroupsAdded: response.user.userGroupsAdded,
+                                                    userGroupsRemoved: response.user.userGroupsRemoved
+                                                }
+                                            }
+                                        })
+                                    );
+                                }
+                            }
+                        });
+
+                    dialogReference.afterClosed().subscribe((response) => {
+                        this.store.dispatch(
+                            selectTenant({
+                                id: request.user.id
+                            })
+                        );
+                    });
+                })
+            ),
+        { dispatch: false }
+    );
+
+    updateUser$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(UserListingActions.updateUser),
+            map((action) => action.request),
+            switchMap((request) =>
+                from(this.usersService.updateUser(request)).pipe(
+                    map((response) =>
+                        UserListingActions.updateUserSuccess({
+                            response: {
+                                user: response,
+                                userGroupUpdate: response.userGroupUpdate
```

Comment:
response does not have a member `userGroupUpdate`. This prevents adding a user to a group when editing the user.
```suggestion
                                userGroupUpdate: request.userGroupUpdate
```